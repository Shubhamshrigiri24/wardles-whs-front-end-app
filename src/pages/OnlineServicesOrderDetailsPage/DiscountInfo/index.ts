export { default, DiscountInfo } from "./DiscountInfo";
