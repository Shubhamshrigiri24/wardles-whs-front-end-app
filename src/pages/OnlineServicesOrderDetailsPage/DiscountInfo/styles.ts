import { makeStyles, Theme } from "@material-ui/core";
import { wellColors } from "@welldigital/components";

export const useStyles = makeStyles((theme: Theme) => {
  return {
    title: {
      color: wellColors.elixir[900],
      fontWeight: theme.typography.fontWeightRegular,
      marginBottom: "5px",
    },
    text: {
      color: wellColors.elixir[400],
      fontWeight: theme.typography.fontWeightRegular,
    },
    promoIcon: {
      marginRight: "12px",
      color: wellColors.elixir[200],
    },
  };
});
