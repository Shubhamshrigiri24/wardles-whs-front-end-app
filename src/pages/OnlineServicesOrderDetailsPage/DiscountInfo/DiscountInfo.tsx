import React, { FC } from "react";
import { Grid, Box } from "@material-ui/core";
import { Typography } from "@welldigital/components";
import LocalOfferIcon from "@material-ui/icons/LocalOffer";
import { useStyles } from "./styles";

export type DiscountInfoProps = {
  title: string;
  label: string;
  price: string;
};

export const DiscountInfo: FC<DiscountInfoProps> = ({
  title,
  label,
  price,
}) => {
  const classes = useStyles();
  return (
    <>
      <Typography className={classes.title} data-testid={"discountInfo/title"}>
        {title}
      </Typography>
      <Grid container justify={"space-between"} wrap={"nowrap"}>
        <Grid item xs>
          <Box display={"flex"} alignItems={"center"}>
            <LocalOfferIcon className={classes.promoIcon} fontSize={"small"} />
            <Typography
              className={classes.text}
              data-testid={"discountInfo/label"}
            >
              {label}
            </Typography>
          </Box>
        </Grid>
        <Grid item xs={"auto"}>
          <Typography
            className={classes.text}
            align={"right"}
            noWrap
            data-testid={"discountInfo/price"}
          >
            {price}
          </Typography>
        </Grid>
      </Grid>
    </>
  );
};

export default DiscountInfo;
