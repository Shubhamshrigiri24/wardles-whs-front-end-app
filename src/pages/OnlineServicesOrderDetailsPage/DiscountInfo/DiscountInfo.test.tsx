import React from "react";
import { render } from "@testing-library/react";
import DiscountInfo, { DiscountInfoProps } from "./DiscountInfo";

const TITLE = "title";
const LABEL = "label";
const PRICE = "Â£3.99";

const makeProps = (props?: Partial<DiscountInfoProps>): DiscountInfoProps => {
  return {
    title: TITLE,
    label: LABEL,
    price: PRICE,
    ...props,
  };
};

describe("<DiscountInfo />", () => {
  it("should match snapshot", () => {
    const props = makeProps();
    const { container } = render(<DiscountInfo {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render props correctly", () => {
    const props = makeProps();
    const { getByTestId } = render(<DiscountInfo {...props} />);

    expect(getByTestId("discountInfo/title")).toHaveTextContent(TITLE);
    expect(getByTestId("discountInfo/label")).toHaveTextContent(LABEL);
    expect(getByTestId("discountInfo/price")).toHaveTextContent(PRICE);
  });
});
