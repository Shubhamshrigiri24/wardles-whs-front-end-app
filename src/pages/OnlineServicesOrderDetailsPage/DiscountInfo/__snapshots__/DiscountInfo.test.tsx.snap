// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscountInfo /> should match snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root makeStyles-title-1 MuiTypography-body1"
    data-testid="discountInfo/title"
  >
    title
  </p>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
    >
      <div
        class="MuiBox-root MuiBox-root-4"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-promoIcon-3 MuiSvgIcon-fontSizeSmall"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.41 11.58l-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58.55 0 1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41 0-.55-.23-1.06-.59-1.42zM5.5 7C4.67 7 4 6.33 4 5.5S4.67 4 5.5 4 7 4.67 7 5.5 6.33 7 5.5 7z"
          />
        </svg>
        <p
          class="MuiTypography-root makeStyles-text-2 MuiTypography-body1"
          data-testid="discountInfo/label"
        >
          label
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto"
    >
      <p
        class="MuiTypography-root makeStyles-text-2 MuiTypography-body1 MuiTypography-noWrap MuiTypography-alignRight"
        data-testid="discountInfo/price"
      >
        Â£3.99
      </p>
    </div>
  </div>
</div>
`;
